include/seastar/core/sstring.hh

```diff
@@ -106,6 +107,7 @@ public:
     struct initialized_later {};
 
     basic_sstring() noexcept {
+        u.external.is_ref = false;
```

Comment:
No, "u" is a union, you need to set either external or internal, you can't set both... In this constructor, the string is internal, so you musn't set external.